bpf/utils.h

```diff
@@ -337,4 +338,129 @@ static inline bool check_and_do_flow_filtering(flow_id *id, u16 flags) {
     return false;
 }
 
+static inline void core_fill_in_l2(struct sk_buff *skb, flow_id *id, u16 *family) {
+    struct ethhdr eth;
+
+    __builtin_memset(&eth, 0, sizeof(eth));
+
+    u8 *skb_head = BPF_CORE_READ(skb, head);
+    u16 skb_mac_header = BPF_CORE_READ(skb, mac_header);
+
+    bpf_probe_read(&eth, sizeof(eth), (struct ethhdr *)(skb_head + skb_mac_header));
+    __builtin_memcpy(id->dst_mac, eth.h_dest, ETH_ALEN);
+    __builtin_memcpy(id->src_mac, eth.h_source, ETH_ALEN);
+    id->eth_protocol = bpf_ntohs(eth.h_proto);
+    if (id->eth_protocol == ETH_P_IP) {
+        *family = AF_INET;
+    } else if (id->eth_protocol == ETH_P_IPV6) {
+        *family = AF_INET6;
+    }
```

Comment:
previously, `family` was obtained that way: `family = BPF_CORE_READ(sk, __sk_common.skc_family);` in code for RTT ... I wonder if this change could impact the bug #382 , where in some situations it seemed like ipv4 aren't correctly read

Reply:
I will check on the issue with RTT once this is merged see if I can repo but its possible if skb family wasn't set it will default to v4 we will see